### PR TITLE
Update semantic highlighting sagemaker endpoint deploy scripts

### PR DESCRIPTION
**Pull request by junqiu-lei**
### Description
This PR update semantic highlighting remote model example scripts to:
* Support Batch inference API with BWC
* Add new example for how to use `modernbert` model with sentence similarity compare strategy.
* Refactor code to adapt above two changes, and to be extensible for other models/strategies.
 
### Issues Resolved
https://github.com/opensearch-project/neural-search/issues/1516
 
### Check List
- [x] New functionality has been documented.
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: docs/source/examples/semantic_highlighting/README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+- **opensearch-semantic-highlighter**: OpenSearch semantic highlighter v1 model, see huggingface [here](https://huggingface.co/opensearch-project/opensearch-semantic-highlighter-v1)
+- **modernbert**: ModernBERT-base model for semantic highlighting, see huggingface [here](https://huggingface.co/answerdotai/ModernBERT-base)
+
+## Usage
+
+### Prerequisites
+
+Install deployment dependencies:
+```bash
+pip install -r requirements.txt
```
[the remark this answers is not in the document]
Comment (junqiu-lei) [reply docs/source/examples/semantic_highlighting/README.md]: Updated to:

```bash
cd docs/source/examples/semantic_highlighting
pip install -r requirements.txt
```

File: docs/source/examples/semantic_highlighting/modernbert/inference.py [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+
+def highlight_sentences(question, context, min_score=0.65) -> List[HighlightSpan]:
+    """
+    Highlight relevant sentences using ModernBERT with normalized scoring.
+    
+    Returns character-level positions of highlighted sentences.
+    """
+    # Split context into sentences
+    sentences = split_sentences(context)
```
Comment (greaa-aws) [inline docs/source/examples/semantic_highlighting/modernbert/inference.py]: This will embed the sentences independently of one another, rather than sharing context across sentences (or between sentence and question).

File: docs/source/examples/semantic_highlighting/modernbert/inference.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    # Calculate cosine similarity
+    scores = torch.nn.functional.cosine_similarity(
+        query_emb.unsqueeze(0),
+        sentence_embs,
+        dim=-1
+    )
+    
+    min_val = scores.min()
+    max_val = scores.max()
+    if torch.isclose(max_val, min_val):
```
Comment (greaa-aws) [inline docs/source/examples/semantic_highlighting/modernbert/inference.py]: Why min-max normalize? The cosine similarity scores are already between 0 and 1, and their magnitudes are informative about the similarity between the question and each sentence. Rescaling these could affect the result.

File: docs/source/examples/semantic_highlighting/modernbert/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,2 @@
+transformers>=4.46.0
```
[the remark this answers is not in the document]
Comment (junqiu-lei) [reply docs/source/examples/semantic_highlighting/modernbert/requirements.txt]: Makes sense, updated to use:
```
transformers>=4.48.0
torch>=2.6.0
```

File: docs/source/examples/semantic_highlighting/modernbert/inference.py [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+
+    query_terms = set(_tokenize_for_overlap(question))
+
+    highlights = []
+    for i in range(len(sentences)):
+        norm_score = normalized[i].item()
+        if norm_score < min_score:
+            continue
+
+        sentence_terms = set(_tokenize_for_overlap(sentences[i]['text']))
```
[the remark this answers is not in the document]
Comment (junqiu-lei) [reply docs/source/examples/semantic_highlighting/modernbert/inference.py]: Removed the token overlap requirement entirely to properly leverage ModernBERT's semantic capabilities.

File: docs/source/examples/semantic_highlighting/modernbert/inference.py [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+def get_embedding(text):
+    """Get sentence embedding from ModernBERT"""
+    inputs = tokenizer(text, return_tensors="pt", truncation=True, max_length=8192).to(DEVICE)
+    with torch.no_grad():
+        outputs = model(**inputs)
+        embedding = outputs.last_hidden_state.mean(dim=1).squeeze()
+    return embedding
+
+
+def split_sentences(text):
```
Comment (greaa-aws) [inline docs/source/examples/semantic_highlighting/modernbert/inference.py]: This is somewhat brittle. If we're already using `nltk.sent_tokenize()` for the sentence highlighter model, then I'd recommend using it here too for more robust sentence splitting.

Comment (junqiu-lei) [reply docs/source/examples/semantic_highlighting/modernbert/inference.py]: Updated to use `nltk.sent_tokenize()`
